addon/views/modal.js

```diff
@@ -33,6 +34,23 @@ export default Em.View.extend(
     this.set('visible', true);
   },
 
+  /* Hide with Escape */
+  turnOnEscapeEvent: function() {
+    var _this = this;
+
+    this.set('keydownListener', function(event) {
```

Comment:
Yes, it was the original implementation i used, but i was worried about the times the event will be registered. (every time the element is inserted)